godrive/auth.go

```diff
@@ -24,6 +24,37 @@ func (s *Server) Auth(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx, span := s.tracer.Start(r.Context(), "auth middleware")
 
+		// Handle Api Token auth
+		if authToken := r.Header.Get(auth.AuthorizationHeader); authToken != "" {
+			apiToken, err := s.db.GetApiToken(ctx, authToken)
+			if err != nil {
+				span.RecordError(err)
+				slog.ErrorContext(ctx, "failed to find api token: %w", slog.Any("err", err))
+				span.End()
+				s.error(w, r, err, http.StatusUnauthorized)
+				return
+			}
+			var info auth.UserInfo
+			user, err := s.db.GetUser(ctx, apiToken.UserID)
+			if err != nil {
+				span.RecordError(err)
+				slog.ErrorContext(ctx, "failed to get user by name: %w", slog.Any("err", err))
+				span.End()
+				s.error(w, r, err, http.StatusInternalServerError)
+				return
+			}
+			info.Home = user.Home
```

Comment:
I'm not really happy with this, feels very brittle and ugly

Reply:
ugly but not avoidable